src/c++/perf_analyzer/genai-perf/README.md

```diff
@@ -61,93 +59,182 @@ all from the [command line](#cli).
 
 ```bash
-RELEASE="24.03"
+export RELEASE="mm.yy" # e.g. export RELEASE="24.03"
 
-pip install "git+https://github.com/triton-inference-server/client.git@r${RELEASE}#egg=genai-perf&subdirectory=src/c++/perf_analyzer/genai-perf"
+pip install "git+https://github.com/triton-inference-server/client.git@r${RELEASE}#subdirectory=src/c++/perf_analyzer/genai-perf"
+```
+
+Run GenAI-Perf:
 
+```bash
 genai-perf --help
 ```
 
-# Basic Usage
+</details>
+</br>
+
+# Quick Start
+
+## Measuring Throughput and Latency of GPT2 using Triton + TensorRT-LLM
 
-## Triton with TRT-LLM
+### Running GPT2 on Triton Inference Server using TensorRT-LLM
+
+<details>
+<summary>See instructions</summary>
+
+1. Run Triton Inference Server with TensorRT-LLM backend container:
 
 ```bash
-genai-perf -m llama-2-7b --concurrency 1 --service-kind triton --backend trtllm
+export RELEASE="mm.yy" # e.g. export RELEASE="24.03"
+
+docker run -it --net=host --rm --gpus=all --shm-size=2g --ulimit memlock=-1 --ulimit stack=67108864 nvcr.io/nvidia/tritonserver:${RELEASE}-trtllm-python-py3
 ```
 
-## Triton with vLLM
+2. Install Triton CLI (~5 min):
 
 ```bash
-genai-perf -m llama-2-7b --concurrency 1 --service-kind triton --backend vllm
+pip install \
+  --extra-index-url https://pypi.nvidia.com \
+  -U \
+  psutil \
+  "pynvml>=11.5.0" \
+  torch==2.1.2 \
+  tensorrt_llm==0.8.0 \
+  "git+https://github.com/triton-inference-server/triton_cli@0.0.6"
 ```
 
-## OpenAI Chat Completions Compatible APIs
+3. Download model:
+
+```bash
+triton import -m gpt2 --backend tensorrtllm
+```
 
-https://platform.openai.com/docs/api-reference/chat
+4. Run server:
 
 ```bash
-genai-perf -m llama-2-7b --concurrency 1 --service-kind openai --endpoint-type chat
+triton start
 ```
 
-## OpenAI Completions Compatible APIs
+</details>
 
-https://platform.openai.com/docs/api-reference/completions
+### Running GenAI-Perf
+
+1. Run Triton Inference Server SDK container:
 
 ```bash
-genai-perf -m llama-2-7b --concurrency 1 --service-kind openai --endpoint-type completions
+export RELEASE="mm.yy" # e.g. export RELEASE="24.03"
+
+docker run -it --net=host --rm --gpus=all nvcr.io/nvidia/tritonserver:${RELEASE}-py3-sdk
 ```
 
-> [!Note]
-> GenAI-Perf uses Llama tokenizer as a default tokenizer to parse and calculate
-> token metrics on the input prompts and output responses. Users can instead
-> specify a custom huggingface tokenizer using `--tokenizer` command line option
-> as well.
+2. Run GenAI-Perf:
+
+```bash
+genai-perf \
+  -m gpt2 \
+  --service-kind triton \
+  --backend trtllm \
+  --prompt-source synthetic \
+  --num-prompts 100 \
+  --random-seed 123 \
+  --synthetic-input-tokens-mean 200 \
+  --synthetic-input-tokens-stddev 0 \
+  --streaming \
+  --output-tokens-mean 100 \
+  --output-tokens-stddev 0 \
+  --output-tokens-mean-deterministic \
+  --tokenizer hf-internal-testing/llama-tokenizer \
+  --concurrency 1 \
+  --measurement-interval 4000 \
+  --profile-export-file my_profile_export.json \
```

Comment:
do we want this option if it was bugged until after 24.04?

Reply:
or is this our current docs and not an update for a release?

Reply:
This is for main branch so since it works in main branch we should showcase it here

Reply:
got it. thought this was for r24.04.
this is good to go! nice work man.